crypto.h

```diff
@@ -99,3 +105,34 @@ NAN_METHOD(getk){
 
   ret(o);
 }
+
+
+//api: tcpsendstr(s, str)
+NAN_METHOD(tcpsendstr){
+  unsigned char msg[MAX_INPUT_LEN]; /* reinterpreted msg, try to reuse later */
+  size_t msz;
+
+  //TODO: deadline control
+  int64_t deadline = -1;
+
+  tcpsock s = UnwrapPointer<tcpsock>(info[0]);
+  utf8 str(info[1]);
+
+  unsigned long long len = reinterpret_str(*str, (unsigned char *)msg);
+  randombytes_buf(&nonce, crypto_box_NONCEBYTES);
```

Comment:
sure, [link1](https://github.com/jedisct1/libsodium/blob/fce6852d64339efa33c0ee4130b3107b888d6067/examples/box.c#L29-L31) and [link2](https://github.com/jedisct1/libsodium/blob/fce6852d64339efa33c0ee4130b3107b888d6067/examples/box.c#L81) :)